src/sage/ext/memory.pyx

```diff
@@ -3,14 +3,14 @@ Low-level memory allocation functions
 
 TESTS:
 
-Check that a ``MemoryError`` is raised if we try to allocate a
+Check that an error is raised if we try to allocate a
 ridiculously large integer, see :trac:`15363`::
 
-    sage: 2^(2^63-3)
-    Traceback (most recent call last):
-    ...
-    OverflowError: exponent must be at most 2147483647         # 32-bit
-    RuntimeError: Aborted                                      # 64-bit
+    sage: try:
+    ....:     2^(2^63-3)
+    ....: except (OverflowError, RuntimeError, FloatingPointError):
+    ....:     print ('Overflow error')
+    ...Overflow error
```

Comment:
It'd be nice to have some explanation for the `...` since it looks like we're already trying to list all of the possible errors and replace them with `'Overflow error'`

Otherwise, the CI is happy now...

Reply:
IIRC the old gmp used to actually print some stuff in addition to an error; in the previous form of the test this would be catched by the `...` of the traceback.

It'd feel better if sagemath would catch all these variations and return always the same exception (say `OverflowError`), then the doctest would document precisely what happens (unlike here where the doctest is documenting that different things can happen but we don't know which one).

I think getting this merged is still a win. Whoever feels inclined can improve this in a separate PR.